GoMeet/src/main/webapp/test/spec/spec-manual-time-proposal.js

```diff
@@ -0,0 +1,325 @@
+const MEETING_TIME_INPUT_DIV_ID = 'meeting-time-inputs';
+
+// Fake 'now': 2020-12-22 11:12
+const YEAR_NOW = 2020;
+const MONTH_NOW = 11; // Note: 0-11 is Jan-Dec
+const DAY_NOW = 22;
+const HOUR_NOW = 11;
+const MINS_NOW = 12;
+const SEC_NOW = 45;
+const MILLISEC_NOW = 112;
+const ISO_DATESTRING = `${YEAR_NOW}-${MONTH_NOW+1}-${DAY_NOW}T${HOUR_NOW}:${MINS_NOW}:00.000`;
+const FAKE_NOW = new Date(YEAR_NOW, MONTH_NOW, DAY_NOW, HOUR_NOW, MINS_NOW, SEC_NOW, MILLISEC_NOW);
+
+function resetPageState() {
+  // reset div to state it was onload (with one child div), so that we get a fresh div to work with next time
+  var inputDiv = document.getElementById(MEETING_TIME_INPUT_DIV_ID);
+  var defaultInputField = document.createElement('div');
+  // re-add the 3 fields that mimic the structure of a div for a time input
+  defaultInputField.innerHTML = '<label></label><input><button></button>';
+  defaultInputField.id = MEETING_TIME_NAME_PREFIX + '1';
+  inputDiv.innerHTML = '';
+  inputDiv.appendChild(defaultInputField);
+
+  // reset enteredTimes
+  enteredTimes.clear();
+
+  // set the global timeInputs and timeNameSuffix back to their original value of 1
+  timeInputs = 1;
+  timeNameSuffix = 1;
+}
+
+// Tests for addTimeInput
+describe('addTimeInput', function () {
+  beforeAll(function() {
+    spyOn(window, 'alert');
+  });
+
+  beforeEach(function() {
+    // mock the Date object
+    jasmine.clock().install()
+    jasmine.clock().mockDate(FAKE_NOW);
+  })
+
+  it('adds one time input to the page with a label and a delete button', function() {
+    addTimeInput(document);
+    var inputName = MEETING_TIME_NAME_PREFIX + timeNameSuffix;
+    var addedInputDiv = document.getElementById(inputName);
+
+    // check that all nodes with appropriate settings have been added
+    expect(addedInputDiv.id).toEqual(inputName);
+    expect(addedInputDiv.children.length).toEqual(3); // three children - <label>, <input> and <button>
+    
+    // The child nodes were appended in the order of <label>, <input>, <button>
+    var label = addedInputDiv.children.item(0);
+    expect(label.tagName).toEqual('LABEL'); // Note: docs say that returned tagname is always upper case
+    expect(label.htmlFor).toEqual(inputName);
+    expect(label.innerText).toEqual('New Time');
+
+    var inputField = addedInputDiv.children.item(1);
+    expect(inputField.tagName).toEqual('INPUT');
+    expect(inputField.type).toEqual('datetime-local');
+    expect(inputField.name).toEqual(inputName);
+    expect(inputField.min).toEqual(ISO_DATESTRING);
+    expect(inputField.onchange.toString()).toBe('function() {rectifyInputtedTime(this);}');
+
+    var deleteButton = addedInputDiv.children.item(2);
+    expect(deleteButton.tagName).toEqual('BUTTON');
+    expect(deleteButton.innerText).toEqual('-');
```

Comment:
optional suggestion: since you care about functionality more than interface, you probably don't need to verify the UI text labels.

Reply:
I've removed the checks, because this would also make it easier for us to change the UI for stylistic purposes later. Thanks for the suggestion!